plugin/file/lookup.go

```diff
@@ -92,6 +92,12 @@ func (z *Zone) Lookup(ctx context.Context, state request.Request, qname string)
 			break
 		}
 
+		if nextElem, found = tr.Next(parts); found {
```

Comment:
Tests fail if the following record is added to the zone: `x.&.d.example.org.        IN      A       127.0.1.53`.

I think it's because of an assumption that a `*` element will always be ordered in the tree before any other element in the same sub domain (and `&` is sorted before `*`).  
E.g. the following record does not cause tests to fail: `x.+.d.example.org.        IN      A       127.0.1.53` presumably because `+` sorts after `*`.

Reply:
sorting and tr.Next don't do what you think. 

and in general I think this PR just happens to implement what the tests hand it